insonmnia/worker/server.go

```diff
@@ -1096,6 +1096,70 @@ func (m *Worker) setupRunningContainers() error {
 	return nil
 }
 
+func (m *Worker) tasksGC() error {
+	tk := time.NewTicker(m.cfg.TasksGCPeriod)
+	for {
+		select {
+		case <-m.ctx.Done():
+			return errors.New("context cancelled")
+		case <-tk.C:
+			log.G(m.ctx).Debug("going to GC containers")
+			if err := m.removeExpiredContainers(); err != nil {
+				log.G(m.ctx).Warn("failed to GC containers", zap.Error(err))
+			}
+		}
+	}
+}
+
+func (m *Worker) removeExpiredContainers() error {
+	dockerClient, err := client.NewEnvClient()
+	if err != nil {
+		return err
+	}
+	defer dockerClient.Close()
+
+	// Collect only running containers.
```

Comment:
because that was the task definition. if we try to gcollect all types of leftover containers, the implementation is going to be much more complicated.